libsql/internal/http/http.go

```diff
@@ -3,61 +3,90 @@ package http
 import (
 	"bytes"
 	"context"
+	"database/sql/driver"
 	"encoding/json"
 	"fmt"
 	"io"
 	"net/http"
+	"sort"
 	"time"
 
 	"github.com/xwb1989/sqlparser"
 )
 
 var httpClient = &http.Client{Timeout: 120 * time.Second}
 
-type params struct {
-	Names  []string
-	Values []any
-}
+type paramsType int
 
-func (p *params) MarshalJSON() ([]byte, error) {
-	if len(p.Values) == 0 {
-		return json.Marshal([]string{})
-	}
-	if len(p.Names) == 0 {
-		return json.Marshal(p.Values)
+const (
+	namedParameters paramsType = iota
+	positionalParameters
+)
```

Comment:
I would remove this all together and just use bool that's true when parameter is a named parameter.

Reply:
My idea with the enum is to make clear to a new developer which are all the type parameters possible. To developers already used to the codebase  have some knowledge with sql it's probably clear. But IMHO, I tend to avoid boolean/functions like `isNameParameters` beause, while I'm read it, I need to put extra effort thinking "what are the other options of parameters besides `named` ones?". With enums, the code tells me what is the opposite of a named parameter. It helps me thinking during code and I believe  it helps news developers also.
What do you think?

Reply:
I generally share your sentiment but I'm not sure how helpful it is in this case. This PR has more serious problem so let's park this one and come back to it once the major problems are fixed.